blog/content/post/2022-12-16-trenchboot-aem-for-qubesos.md

```diff
@@ -0,0 +1,367 @@
+the platform and perform TXT launch. Most of the work for that particular part
+has been done by [Oracle Team working on TrenchBoot for GRUB](https://www.mail-archive.com/grub-devel@gnu.org/msg30167.html).
+That work however covered the Linux kernel TXT launch only. What still had to
+be done was the multiboot protocol support in GRUB to be able to TXT launch a
+Xen Hypervisor. The patches have been prepared for the respective Qubes GRUB
+[package](https://github.com/3mdeb/qubes-grub2/pull/2).
+
+## Xen modifications
+
+![](/img/xen_project_logo.png)
+
+Analogically to GRUB, Xen had to take over some responsibilities from tboot.
+Due to the Intel TXT requirements for boot process, a new entry point had to be
+developed to which SINIT ACM will return control. The new entry point was
+responsible for saving information that a TXT launch happened and clean up the
+processor state so that the booting of Xen kernel could continue with the
+standard multiboot path. Among others, if Xen detected TXT launch, it had to
+perform the special processor cores wakeup process (which has been rewritten
+from TrenchBoot Linux patches to Xen native code) and measure external
+components before using them (that is the Xen parameters, Dom0 Linux kernel and
+initrd). Xen also had to reserve the memory regions used by Intel TXT, like it
+was done when tboot has been utilized. The relevant source code for the
+respective Qubes Xen package is available [here](https://github.com/3mdeb/qubes-vmm-xen/pull/1).
+
+# Installation and verification of TrenchBoot AEM on Qubes OS
+
+For a seamless deployment and installation of TrenchBoot AEM, the modifications
+have been converted to patches which are applied to projects' sources during
+Qubes OS components compilation. Those patches have been presented earlier with
+links to Pull Requests. It allows building ready-to-use RPM packages that can
+be installed directly on an installed Qubes OS system. Below a procedure for
+building the packages has been presented. If your are not interested in
+compilation, skip to the [next section](#installing-xen-and-grub-packages).
+The pre-built packages can be downloaded from [here](https://3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_poc/).
+The packages have been covered with SHA512 sums signed with 3mdeb's
+`Qubes OS TrenchBoot AEM open-source software release 0.x signing key`
+available on [3mdeb-secpack repository](https://github.com/3mdeb/3mdeb-secpack/blob/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc).
+To verify the RPM packages, fetch the key with:
+`gpg --fetch https://raw.githubusercontent.com/3mdeb/3mdeb-secpack/master/open-source-software/qubes-os-trenchboot-aem-open-source-software-release-0.x-signing-key.asc`
+and then for each package please run:
+
+```txt
+gpg --verify <rpm_package-name>.sha512.sig <rpm_package-name>.sha512
+sha512sum -c <rpm_package-name>.sha512
```

Comment:
Wouldn't it be easier to have a single signed sha512sums file?

Reply:
Probably yes... I will do that.

Reply:
made a single sha512susm file with a single signature
https://3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_poc/sha512sums
https://3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_poc/sha512sums.sig